core/src/cell/node.rs

```diff
@@ -196,16 +196,19 @@ impl LocalNode {
             ..Default::default()
         };
 
-        Self::new_from_config(config).expect("Couldn't create node config generated config")
+        Self::new_from_config(&config).expect("Couldn't create node config generated config")
     }
 
-    pub fn new_from_config(config: LocalNodeConfig) -> Result<Self, Error> {
+    pub fn new_from_config(config: &LocalNodeConfig) -> Result<Self, Error> {
         let keypair = Keypair::decode_base58_string(&config.keypair)
             .map_err(|err| Error::Cell(format!("Couldn't decode local node keypair: {}", err)))?;
 
         let node = LocalNode {
             node: Node::new_from_public_key(keypair.public()),
-            identity: Arc::new(LocalNodeIdentity { keypair, config }),
+            identity: Arc::new(LocalNodeIdentity {
+                keypair,
+                config: config.clone(),
```

Comment:
So, since we're cloning here, I'd actually revert the method to receive a `LocalNodeConfig`, not a reference. The reason is that the caller of the method may have just created an instance of `LocalNodeConfig` before calling this method, and then it gets cloned here again, we will have created in total 2 instance instead of only 1.

Same for all the other places you switched to a reference instead.